### PR TITLE
Change installation directory and add dotnet-quality input

### PR DESCRIPTION
**Pull request by IvanZosimov**
**Description:**
In the scope of this PR, installation directories for Linux and Windows were
changed as well as the logic for resolving `dotnet-version` was updated
and a new input `dotnet-quality` was added.

**Related issue:**
https://github.com/actions/runner-images-internal/issues/4058

**Check list:**
- [x] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,52 +114,58 @@ steps:
...
     NUGET_AUTH_TOKEN: ${{secrets.AZURE_DEVOPS_PAT}} # Note, create a secret with this name in Settings
 - name: Publish the package to Azure Artifacts
   run: dotnet nuget push <my project>/bin/Release/*.nupkg
+```
 
-# Authenticates packages to push to nuget.org.
-# It's only the way to push a package to nuget.org feed for macOS/Linux machines due to API key config store limitations.
-- uses: actions/setup-dotnet@v2
+### nuget.org
+```yml
```
Comment (dmitry-shibanov) [inline README.md:133]: Do we need to delete hint about authentication ?

File: __tests__/installer.test.ts [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
     process.env.DOTNET_ROOT = '';
-    await io.rmRF(toolDir);
-    await io.rmRF(tempDir);
-  });
+    try {
+      await io.rmRF(`${toolDir}/*`);
+      await io.rmRF(`${tempDir}/*`);
+    } catch (err) {
+      console.log(err.message);
+      console.log('Failed to remove test directories');
```
Comment (dmitry-shibanov) [inline __tests__/installer.test.ts]: I think we can combine them.

File: __tests__/installer.test.ts [hunk 3]
```diff
@@ -1,33 +1,48 @@
...
+  afterEach(async () => {
     try {
-      await io.rmRF(toolDir);
-      await io.rmRF(tempDir);
-    } catch {
+      await io.rmRF(`${toolDir}/*`);
+      await io.rmRF(`${tempDir}/*`);
+    } catch (err) {
+      console.log(err.message);
       console.log('Failed to remove test directories');
```
Comment (dmitry-shibanov) [inline __tests__/installer.test.ts]: Here is the same

File: README.md (hunk 1 above)
Comment (IvanZosimov) [reply README.md:133]: It just looks like it's deleted, but it's in line 140 under the `>**Note**` tag.

**Review by marko-zivic-93 — APPROVED** [review]

**Review by panticmilos — APPROVED** [review]

**Review by e-korolevskii — APPROVED** [review]

File: src/installer.ts [hunk 4]
```diff
@@ -2,187 +2,159 @@
...
+        this.resolvedArgument.type = 'channel';
+        const httpClient = new hc.HttpClient('actions/setup-dotnet', [], {
+          allowRetries: true,
+          maxRetries: 3
+        });
+        this.resolvedArgument.value = await this.getReleasesJsonUrl(
+          httpClient,
+          [major, minor]
+        );
+      }
```
Comment (e-korolevskii) [inline src/installer.ts]:
```suggestion
if (this.testTag(major)) {
        this.resolvedArgument.type = "channel";
        if (this.testTag(minor)) {
          this.resolvedArgument.value = `${major}.${minor}`;
        } else {
          this.resolvedArgument.type = "channel";
          const httpClientOptions = {
            allowRetries: true,
            maxRetries: 3,
          };
          const httpClient = new hc.HttpClient(
            "actions/setup-dotnet",
            [],
            httpClientOptions
          );
          this.resolvedArgument.value = await this.getReleasesJsonUrl(
            httpClient,
            [major, minor]
          );
        }
      }
```

Comment (IvanZosimov) [reply src/installer.ts]: Thanks, good suggestion! Looks better now 😉

**Review by marko-zivic-93 — APPROVED** [review]

**Comment by IvanZosimov** [comment]
Close this draft PR as actual was created https://github.com/actions/setup-dotnet/pull/315

**Pull request closed without merging**